### ISSUE TITLE
Cody codebase doesn't detect the forked repo name

### ISSUE THREAD

**Issue by deepak2431** [sourcegraph/sourcegraph#51268]
As Cody 0.0.9 version has been released, which has the new features to autodetect codebase from the git clone URL, but it missed these two cases when it should work to detect the URL:
1. It doesn't detect the codebase name for the forked repo clone.
2. Many users have multiple Github accounts, so their GitHub hostnames are different per the ssh config.

#### Expected behaviour:
It should detect codebase settings for these cases too.

#### Actual behaviour:
It just says the codebase was not detected for the forked repo clone with a different hostname than Github.

**Comment by abeatrix** [comment]
Hey @deepak2431 thanks for bringing this to our attention! 

Does the changes in this PR covers your issue here? https://github.com/sourcegraph/sourcegraph/pull/51274

**Comment by deepak2431** [comment]
> Hey @deepak2431 thanks for bringing this to our attention!
> 
> Does the changes in this PR covers your issue here? #51274

Looking into the tests, I don't think it covers the case. As if I have forked a repo with my username, the git origin URL would be something like git@github.com/deepak2431/sourcegraph.git

**Comment by abeatrix** [comment]
Makes sense! In this case you might need to set the config Cody.codebase manually, which is broken in the current version v0.0.9 because if a git url is detected, it would use that automatically instead of the one in the setting, which is now fixed in the linked PR where we will choose the one from setting over the one from git url. 

I hope this helps! (Or if you have other suggestions please let us know!) 🙇‍♀️

**Comment by deepak2431** [comment]
Yeah, got it. I too, got a sense of like something was wrong in the codebase settings config while using the new version, as it was most of the time falling to use the keyword context. Good to see that your PR fixes this and handle most of the different open repository host to get the codebase name.

For the case I mentioned, it's completely fine to set it up manually. But if you think most of the open-source contributors working on an open-source project have a forked clone in their local. So, what do you think of handling this case too? 
What I can think of as a draft version of the solution is using spawnSync to get the username of the git config and checking if it's present in the URL, which means it's a forked clone, and then we can handle this case accordingly by tweaking the current implementation a bit.

**Comment by toolmantim** [comment]
@deepak2431 I just found this while working on the `codebase` setting, and was curious to know how you'd think we'd be able to automatically infer the upstream repo from the fork? Because there wouldn't necessarily be anything present in the local git config pointing to the upstream.

**Comment by deepak2431** [comment]
@toolmantim For me, what I can think of as a solution now is using the `git remote -v | grep fetch` command to list all the remotes configured. It would have both the forked and upstream. See this output: 
![iScreen Shoter - Code - 230531120950](https://github.com/sourcegraph/sourcegraph/assets/44617923/722b1640-cc30-4cab-a966-0b0666e589b5)

Using this output, we can pass this to the existing implemented code to verify which is a valid git URL. Also, if you see the forked repo name is something like: "[git@github.com](mailto:git@github.com)/deepak2431/sourcegraph.git". If we split this into a list using separator "\/", the value at the last two indexes would differ. So, this can also indicate that this configured remote is a forked one.

This would be my initial approach to handle this, but yeah, while implementing, there can be other cases too, which we can handle.

Let me know what you think of this!

**Comment by toolmantim** [comment]
Super, thanks @deepak2431! I suppose you're right that most people working locally on a fork would have the upstream in their remotes. And I'd love to see us provide a great experience for someone working on a fork of an open source project.

To me it sounds problematic to try to automatically detect which origin might be the official upstream using a string heuristic? Or maybe I'm missing something still.

Another option might be to handle it with some nice UI. If we detect "not indexed", then we could show a "Choose one of your git remotes to use instead" UI.

I'm working on the design for these states now, so I'll keep this in mind!

But if you've got other ideas, don't be afraid of sharing them here. Thanks again @deepak2431!

**Comment by deepak2431** [comment]
@toolmantim You're right about the string heuristic. It wouldn't be the better way to do this. 

The following solution you mentioned using the UI sounds much better to me. Also, I would suggest adding a better tooltip-like message on the "INDEXED/NOT INDEXED" so it would be much more transparent for users when to use the Git options and when not.

I am looking forward to the designs of it.
